src/dali_executor/io_descriptor.h

```diff
@@ -28,169 +28,37 @@
 
 namespace triton { namespace backend { namespace dali {
 
-struct StorageCpu {};
-struct StorageGpu {};
-
-namespace detail {
-
-
-template<typename StorageBackend = StorageCpu>
-void allocate(void** ptr, size_t size) {
-  *ptr = malloc(size);
-  if (!*ptr) {
-    throw std::bad_alloc();
-  }
-}
-
-
-template<>
-inline void allocate<StorageGpu>(void** ptr, size_t size) {
-  CUDA_CALL_GUARD(cudaMalloc(ptr, size));
-}
-
-
-template<typename StorageBackend = StorageCpu>
-void deallocate(void* ptr) {
-  free(ptr);
-}
-
-
-template<>
-inline void deallocate<StorageGpu>(void* ptr) {
-  CUDA_CALL_GUARD(cudaFree(ptr));
-}
-
-
-template<typename StorageBackend = StorageCpu>
-void copy(void* dst, const void* src, size_t count) {
-  memcpy(dst, src, count);
-}
-
-
-template<>
-inline void copy<StorageGpu>(void* dst, const void* src, size_t count) {
-  CUDA_CALL_GUARD(cudaMemcpy(dst, src, count, cudaMemcpyDefault));
-}
-}  // namespace detail
-
 template<typename T>
-class IODescriptorBase {
- public:
-  std::string name;
-  dali_data_type_t type;
+struct BufferDescr {
   device_type_t device;
   int device_id;
-  TensorListShape<> shape;
-  span<T> buffer;
-};
+  T* data;
+  size_t size;
```

Comment:
Does this type need to be a POD? If not, I'd strongly recommend adding field initializers. Leaving the fields uninitialized can lead to failures when a new field is added. We've seen this exact kind of failure in DALI before, so it's not theoretical.

Reply:
fixed BufferDescr constructor